### PR TITLE
General Updates 11

### PR DESCRIPTION
**Pull request by DeveloperWOW64**
### Before you start

- [x] Have you checked if a similar PR has already been requested? 
- [x] Have you built and ran the app?

###### <sup>A list of assets (screenshots, mockups) relevant to this pull request.</sup>

Settings Page:

image

### CODE REVIEW

**Comment by DeveloperWOW64** [comment]
@onein528 How did you format [these](https://github.com/FluentHub/FluentHub/tree/main/src/FluentHub/Assets/Icons) icons?

**Comment by onein528** [comment]
He said "Power Automate + ImageMagick" in https://github.com/FluentHub/FluentHub/pull/105#issuecomment-1080785436

**Comment by onein528** [comment]
I can get icons from https://icons8.com/icon/set/user/fluency

**Comment by DeveloperWOW64** [comment]
@onein528 I've made all the images fluent but they're not showing up here:
image

Is this because of Octions? And if so, how do you generate them?

**Comment by onein528** [comment]
> Is this because of Octions? And if so, how do you generate them?

Yes. generated by [icomoon](https://icomoon.io/). and adjusted with FontForge(fixed some path directions)

**Comment by onein528** [comment]
from here https://github.com/primer/octicons/tree/main/icons

**Comment by DeveloperWOW64** [comment]
How exactly do I generate them though

**Comment by onein528** [comment]
Steps:

1. Download icon svgs from https://github.com/primer/octicons/tree/main/icons
2. Click Import icons on https://icomoon.io/app/#/select
image
3. Click Generate Font on the bottom of navigation item

**Comment by onein528** [comment]
but the artifacts generated by icomoon have wrong legal notice and font details/meta. so fix with FontForge.

**Comment by DeveloperWOW64** [comment]
> but the artifacts generated by icomoon have wrong legal notice and font details/meta. so fix with FontForge.

Good - I need to have dinner now so goodnight - Ill continue in the morning

**Comment by DeveloperWOW64** [comment]
> Steps:
> 
> 1. Download icon svgs from https://github.com/primer/octicons/tree/main/icons
> 2. Click Import icons on https://icomoon.io/app/#/select
> 3. Click Generate Font on the bottom of navigation item

How does this relate to the repo, orgs, issues, prs and others?

**Comment by onein528** [comment]
> How does this relate to the repo, orgs, issues, prs and others

What do you mean?

**Comment by DeveloperWOW64** [comment]
> What do you mean?

Sorry, Im very confused. On the homepage, is it just a symbol on top of a colour?

**Comment by onein528** [comment]
Yes. not using `<Image/>` just `<Border><FontIcon/><Border/>`

**Comment by onein528** [comment]
image

**Comment by DeveloperWOW64** [comment]
> Yes. not using `<Image/>` just `<Border><FontIcon/><Border/>`

Okay. Thanks for the info. Are the octions used in any other places?

**Comment by onein528** [comment]
Many places

https://github.com/FluentHub/FluentHub/search?q=Octions

**Comment by DeveloperWOW64** [comment]
@onein528
image

**Comment by onein528** [comment]
Incredible!!

**Comment by DeveloperWOW64** [comment]
> Incredible!!

Thanks! I'm trying to put the SeeMoreActivities Button inside the activities expander but it doesn't work - do you know how?

**Comment by onein528** [comment]
```DIFF
 <muxc:Expander.Content>
     <Grid>
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto" />
+            <RowDefinition Height="Auto" />
+            <RowDefinition Height="*" />
+        </Grid.RowDefinitions>

         <ListView
             Margin="-4,0"
             HorizontalAlignment="Stretch"
             VerticalAlignment="Stretch"
             IsItemClickEnabled="True"
             IsRightTapEnabled="True"
             ItemContainerTransitions="{x:Null}"
             ItemsSource="{x:Bind ViewModel.UserNotifications, Mode=OneWay}"
             SelectionMode="None">
         </ListView>

+        <Button/>
     </Grid>
 </muxc:Expander.Content>
```

Cannot like this?

**Comment by DeveloperWOW64** [comment]
> Cannot like this?

Oh thanks; that works!

**Comment by DeveloperWOW64** [comment]
@onein528 When I go to the code preview page on gu11 i get nullreferenceexception. Can you debug + investigate?

**Comment by onein528** [comment]
@DeveloperWOW64 Please take a screenshot of the previous page

**Comment by DeveloperWOW64** [comment]
> @DeveloperWOW64 Please take a screenshot of the previous page

WDYM?

**Comment by onein528** [comment]
What is a code preview page?

**Comment by DeveloperWOW64** [comment]
> What is a code preview page?

The repositories settings page

**Comment by onein528** [comment]
Okay! but should not happen that. what is name of value caused NullRef?

**Comment by DeveloperWOW64** [comment]
> Okay! but should not happen that. what is name of value caused NullRef?

image
navigationService

**Comment by onein528** [comment]
It is difficult to check out to GU11 due to the breaking changes of the entire repository pages.

**Comment by onein528** [comment]
OK figured out.

**Comment by onein528** [comment]
Add `navigationService = App.Current.Services.GetRequiredService<INavigationService>();` to that page's constructor

**Comment by DeveloperWOW64** [comment]
> Add `navigationService = App.Current.Services.GetRequiredService<INavigationService>();` to that page's constructor

Throws these errors:

image

**Comment by onein528** [comment]
using FluentHub.Core;
using FluentHub.Models;
using FluentHub.Octokit.Models;
using FluentHub.Octokit.Queries.Repositories;
using FluentHub.ViewModels.UserControls.Labels;
using Humanizer;
using Microsoft.Toolkit.Mvvm.ComponentModel;
using Microsoft.Toolkit.Mvvm.Input;
using Microsoft.Toolkit.Mvvm.Messaging;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Threading;
using System.Threading.Tasks;
using Windows.UI.Xaml.Media;
using Windows.UI.Xaml;

replace

**Comment by DeveloperWOW64** [comment]
image

**Comment by DeveloperWOW64** [comment]
It's very confusing

**Comment by onein528** [comment]
add `using Microsoft.Extensions.DependencyInjection;`

**Comment by DeveloperWOW64** [comment]
@onein528 All the imports you're telling me to add are not adding and are unnecessary

**Comment by onein528** [comment]
opps. 

image

try this

using FluentHub.Services;
using FluentHub.ViewModels.AppSettings;
using Microsoft.Extensions.DependencyInjection;
using Windows.UI.Xaml.Controls;
using Windows.UI.Xaml.Navigation;
using muxc = Microsoft.UI.Xaml.Controls;

**Comment by DeveloperWOW64** [comment]
@onein528 I fixed it by adding
```
var provider = App.Current.Services;
navigationService = provider.GetRequiredService<INavigationService>();
```
to public CodePreviewPage()

**Comment by BobbyESP** [comment]
image
Ahhhh meeenn. I'm becoming crazy with thiss. I can't login in the app. It crashes just getting opened

**Comment by DeveloperWOW64** [comment]
> Ahhhh meeenn. I'm becoming crazy with thiss. I can't login in the app. It crashes just getting opened

Are you on main @BobbyESP

**Comment by BobbyESP** [comment]
> > Ahhhh meeenn. I'm becoming crazy with thiss. I can't login in the app. It crashes just getting opened
> 
> Are you on main @BobbyESP 

I switched to gu11, but don't worry, i'll clone the repo again

**Comment by BobbyESP** [comment]
Well, still doesn't working

**Comment by BobbyESP** [comment]
image
I have this error before the one that i've told you

**Comment by BobbyESP** [comment]
It's related to the localizations

**Comment by onein528** [comment]
yes. this exception should be occur. because we have no localization resources. need to delete GetLocalized()

**Comment by BobbyESP** [comment]
Okey no problem!

**Comment by BobbyESP** [comment]
Okay, i modified some lines and the error that i have is because of localizations

**Comment by BobbyESP** [comment]
image
Okay hahah. I think that it's because i have galician as default localization language

**Comment by onein528** [comment]
> image
> 
> I have this error before the one that i've told you

I dont understand why this exception happen. this branch did not change this class.

**Comment by DeveloperWOW64** [comment]
@onein528 When I get these tasks done I'll put GU11 as ready for review
- [ ] Resolve conflicts with GU10 - Use FocusState to change titlebar when clicked
- [ ] Make Lightbulb icon in discussions page Fluent
- [ ] Replace the ToggleSwitch in AppSettings.CodePreviewPage with a ComboBox

**Comment by onein528** [comment]
I see.

**Comment by onein528** [comment]
Good work!! now, normally textbox shows friendly url but when user click that textbox, shows real url!!

Idk what you changed in this pr, so maybe there will be more conflicts after merged gu10.

**Comment by DeveloperWOW64** [comment]
> Good work!! now, normally textbox shows friendly url but when user click that textbox, shows real url!!
> 
> Idk what you changed in this pr, so maybe there will be more conflicts after merged gu10.

Don't think so - I'll compare

**Comment by DeveloperWOW64** [comment]
https://github.com/FluentHub/FluentHub/compare/add-multiaccount-support...gu11
Not looking very good.. 
@onein528 Maybe we can merge gu11 first and see what needs changing? I am okay merging gu11 noew

**Comment by onein528** [comment]
ok!

**Comment by DeveloperWOW64** [comment]
> ok!

@onein528 Can you review?

**Comment by onein528** [comment]
not just change branch name, change commit message to General Updates 10? whichever is fine. your choice

**Comment by onein528** [comment]
And notice.

When merge, delete sub commit message. always:

image

TO

image

**Comment by DeveloperWOW64** [comment]
> not just change branch name, change commit message to General Updates 10? whichever is fine. your choice

WDYM?

**Comment by onein528** [comment]
Do you change GU 10 this?

File: src/FluentHub/Views/Home/NotificationsPage.xaml [hunk 1]
```diff
@@ -57,11 +57,6 @@
                         Foreground="#539bf5"
                         Text="{x:Bind ViewModel.UnreadCount, Mode=OneWay}" />
 
-                    <TextBlock
-                        VerticalAlignment="Center"
-                        Foreground="#539bf5"
-                        Text="new notification" />
```
Comment (onein528) [inline src/FluentHub/Views/Home/NotificationsPage.xaml:63]: Why deleted?

**Comment by DeveloperWOW64** [comment]
> Do you change GU 10 this?

Sorry - I'm really confused - What do you mean?

**Review by onein528 — APPROVED** [review]
LGTM.

**Comment by onein528** [comment]
If you merge this earlier, GU10 will be committed later, causing a number inconsistency.

File: src/FluentHub/Views/Home/NotificationsPage.xaml (hunk 1 above)
Comment (DeveloperWOW64) [reply src/FluentHub/Views/Home/NotificationsPage.xaml:63]: it causes a grammar error

**Comment by DeveloperWOW64** [comment]
> If you merge this earlier, GU10 will be committed later, causing a number inconsistency.

True - it's not a big deal though

**Comment by DeveloperWOW64** [comment]
should i merge or do you really care about the number consistency? I can change it if you do

**Comment by onein528** [comment]
No, just can change this
not needed to change branch name.

image

File: src/FluentHub/Views/Home/NotificationsPage.xaml (hunk 1 above)
Comment (onein528) [reply src/FluentHub/Views/Home/NotificationsPage.xaml:63]: Good.

**Comment by onein528** [comment]
Alright. ready to merge!

**Comment by onein528** [comment]
i'll merge?

**Comment by DeveloperWOW64** [comment]
> i'll merge?

Sure

**Comment by onein528** [comment]
Thank you for your contribution!!

**Comment by DeveloperWOW64** [comment]
> And this is one of the my rules; when the fh member merge, delete sub commit message. always like this:

I don't have access to merge commits. I can only squash and merge

**Comment by onein528** [comment]
Me too. why? need others?

**Comment by DeveloperWOW64** [comment]
> Me too(I restricted). why? need others?

Oh sorry I got confused

**Pull request merged**